lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -46,6 +188,7 @@ def initialize(cmd, debug)
         ps_args = ['-File', self.class.init_path, "\"#{named_pipe_name}\""]
         ps_args << '"-EmitDebugOutput"' if debug
         # @stderr should never be written to as PowerShell host redirects output
+        # TODO: take a fd and turn it into a handle
         stdin, @stdout, @stderr, @ps_process = Open3.popen3("#{cmd} #{ps_args.join(' ')}")
```

Comment:
A better way to do this on Windows would be to start `powershell.exe` through `Process.create` which is monkey-patched by `win32-process` gem, allowing immediate access to the real handle.

However, that puts the module back into the mode of supporting multiple OSes differently, which is less than ideal.  (`popen3` uses Rubys `Process.spawn` under the hood)